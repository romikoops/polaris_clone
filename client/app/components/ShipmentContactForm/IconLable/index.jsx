import React from 'react'
import { gradientTextGenerator } from '../../../helpers'
import PropTypes from '../../../prop-types'
import styles from '../ShipmentContactForm.scss'

export default function IconLable ({ theme, faClass }) {
  const gradientBackground = gradientTextGenerator(theme.colors.primary, theme.colors.secondary)
  return (
    <i className={`${styles.icon_label} fa fa-${faClass} clip`} style={gradientBackground} />
  )
}

IconLable.propTypes = {
  theme: PropTypes.theme,
  faClass: PropTypes.string
}

IconLable.defaultProps = {
  theme: null,
  faClass: ''
}
