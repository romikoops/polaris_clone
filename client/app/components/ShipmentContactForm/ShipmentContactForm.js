import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './ShipmentContactForm.scss';
// import {v4} from 'node-uuid';
import { RoundButton } from '../RoundButton/RoundButton';
import defs from '../../styles/default_classes.scss';
import GmapsWrapper from '../../hocs/GmapsWrapper';
import { PlaceSearch } from '../Maps/PlaceSearch';
import Formsy from 'formsy-react';
import FormsyInput from '../FormsyInput/FormsyInput';

export class ShipmentContactForm extends Component {
    constructor(props) {
        super(props);
        this.state = {
            contactData: props.contactData
        };
        this.handleFormChange = this.handleFormChange.bind(this);
        this.handlePlaceChange = this.handlePlaceChange.bind(this);
        this.handleInvalidSubmit = this.handleInvalidSubmit.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    componentWillReceiveProps(nextProps) {
        this.setState({
            contactData: nextProps.contactData
        });
    }

    handleFormChange(event) {
        this.props.handleChange(event);
    }
    handlePlaceChange(place) {
        const newLocation = {
            streetNumber: '',
            street: '',
            zipCode: '',
            city: '',
            country: '',
            fullAddress: ''
        };
        place.address_components.forEach(ac => {
            if (ac.types.includes('street_number')) {
                newLocation.streetNumber = ac.long_name;
            }

            if (ac.types.includes('route') || ac.types.includes('premise')) {
                newLocation.street = ac.long_name;
            }

            if (ac.types.includes('administrative_area_level_1') || ac.types.includes('locality')) {
                newLocation.city = ac.long_name;
            }

            if (ac.types.includes('postal_code')) {
                newLocation.zipCode = ac.long_name;
            }

            if (ac.types.includes('country')) {
                newLocation.country = ac.long_name;
            }
        });
        newLocation.latitude = place.geometry.location.lat();
        newLocation.longitude = place.geometry.location.lng();
        newLocation.fullAddress = place.formatted_address;
        newLocation.geocodedAddress = place.formatted_address;
        this.setState({
            contactData: { ...this.state.contactData, location: newLocation }
        });
        this.setState({
            autocomplete: { ...this.state.autocomplete, location: true }
        });
    }
    handleSubmit(contactData) {
        this.props.setContact(contactData);
        this.refs.contactForm.reset();
    }
    handleInvalidSubmit() {
        console.log('invalid');
    }
    mapInputs(inputs) {
        const location = {};
        const contact = {};

        for(const k of Object.keys(inputs)) {
            if (k.split('-')[0] === 'location') {
                location[k.split('-')[1]] = inputs[k];
            } else {
                contact[k] = inputs[k];
            }
        }

        return { location, contact };
    }
    render() {
        const { theme } = this.props;
        const { contactData } = this.state;

        return (
            <div className="flex-100 layout-row layout-wrap layout-align-center-start">
                <div className={`flex-none ${defs.content_width} layout-row layout-wrap`}>
                    <Formsy
                        className="flex-100 layout-row layout-wrap layout-align-start-start"
                        name="form"
                        onValidSubmit={this.handleSubmit}
                        onInvalidSubmit={this.handleInvalidSubmit}
                        mapping={this.mapInputs}
                        ref="contactForm"
                        style={{ padding: '8px 20px' }}
                    >
                        <h3>Basic Details</h3>
                        <FormsyInput
                            wrapperClassName="flex-95"
                            className={styles.input}
                            type="text"
                            value={contactData.contact.companyName}
                            name="companyName"
                            placeholder="Company Name"
                        />
                        <FormsyInput
                            wrapperClassName="flex-45"
                            className={styles.input}
                            type="text"
                            value={contactData.contact.firstName}
                            name="firstName"
                            placeholder="First Name"
                        />
                        <FormsyInput
                            wrapperClassName="offset-5 flex-45"
                            className={styles.input}
                            type="text"
                            value={contactData.contact.lastName}
                            name="lastName"
                            placeholder="Last Name"
                        />
                        <FormsyInput
                            wrapperClassName="flex-45"
                            className={styles.input}
                            type="text"
                            value={contactData.contact.email}
                            name="email"
                            placeholder="Email"
                        />
                        <FormsyInput
                            wrapperClassName="offset-5 flex-45"
                            className={styles.input}
                            type="text"
                            value={contactData.contact.phone}
                            name="phone"
                            placeholder="Phone"
                        />
                        <div className="flex-100 layout-row layout-wrap">
                            <h3 className="flex-40">Address Details</h3>
                            <div className="offset-5 flex-55">
                                <GmapsWrapper
                                    theme={theme}
                                    component={PlaceSearch}
                                    inputStyles={{
                                        width: '96%',
                                        marginTop: '9px',
                                        background: 'white'
                                    }}
                                    handlePlaceChange={this.handlePlaceChange}
                                    hideMap={true}
                                />
                            </div>
                        </div>
                        <FormsyInput
                            wrapperClassName="flex-75"
                            className={styles.input}
                            type="text"
                            value={contactData.location.street}
                            name="location-street"
                            placeholder="Street"
                        />
                        <FormsyInput
                            wrapperClassName="offset-5 flex-15"
                            className={styles.input}
                            type="text"
                            value={contactData.location.streetNumber}
                            name="location-streetNumber"
                            placeholder="Number"
                        />
                        <FormsyInput
                            wrapperClassName="flex-25"
                            className={styles.input}
                            type="text"
                            value={contactData.location.zipCode}
                            name="location-zipCode"
                            placeholder="Postal Code"
                        />
                        <FormsyInput
                            wrapperClassName="offset-5 flex-30"
                            className={styles.input}
                            type="text"
                            value={contactData.location.city}
                            name="location-city"
                            placeholder="City"
                        />
                        <FormsyInput
                            wrapperClassName="offset-5 flex-30"
                            className={styles.input}
                            type="text"
                            value={contactData.location.country}
                            name="location-country"
                            placeholder="Country"
                        />
                        <FormsyInput
                            wrapperClassName="flex-100"
                            className={styles.hide}
                            type="text"
                            value={contactData.location.geocodedAddress}
                            name="location-geocodedAddress"
                            placeholder=""
                        />
                        <RoundButton
                            text={`Set ${contactData.type}`}
                            theme={theme}
                            size="small"
                            active
                        />
                    </Formsy>
                </div>
            </div>
        );
    }
}
ShipmentContactForm.propTypes = {
    theme: PropTypes.object,
    shipmentData: PropTypes.object,
    user: PropTypes.object,
    handleChange: PropTypes.func,
    toggleAddressBook: PropTypes.func
};
