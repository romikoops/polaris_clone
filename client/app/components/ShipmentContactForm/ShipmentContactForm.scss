:local {
  .input {
    margin: 10px 0;
    height: 36px;
    border: 1px solid #F2F2F2;
    border-radius: 3px;
    background-color: #F9F9F9;
    box-shadow: 0 2px 3px 0 rgba(237,234,234,0.5);
    width: 100%;
  }
  .icon_btn {
    height: 25px;
    width: 25px;
  }
  .login_form {
    width: 100%;
  }
}