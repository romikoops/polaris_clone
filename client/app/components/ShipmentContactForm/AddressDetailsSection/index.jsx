import React, { PureComponent } from 'react'
import { withNamespaces } from 'react-i18next'
import PropTypes from '../../../prop-types'
import GmapsWrapper from '../../../hocs/GmapsWrapper'
import FormsyInput from '../../FormsyInput/FormsyInput'
import styles from '../ShipmentContactForm.scss'
import PlaceSearch from '../../Maps/PlaceSearch'
import IconLable from '../IconLable'

class AddressDetailsSection extends PureComponent {
  constructor (props) {
    super(props)
    this.state = {
      contactData: props.contactData
    }
  }
  handlePlaceChange (place) {
    const newLocation = {
      streetNumber: '',
      street: '',
      zipCode: '',
      city: '',
      country: ''
    }
    place.address_components.forEach((ac) => {
      if (ac.types.includes('street_number')) {
        newLocation.streetNumber = ac.long_name
      }

      if (ac.types.includes('route') || ac.types.includes('premise')) {
        newLocation.street = ac.long_name
      }

      if (ac.types.includes('administrative_area_level_1') || ac.types.includes('locality')) {
        newLocation.city = ac.long_name
      }

      if (ac.types.includes('postal_code')) {
        newLocation.zipCode = ac.long_name
      }

      if (ac.types.includes('country')) {
        newLocation.country = ac.long_name
      }
    })
    newLocation.latitude = place.geometry.location.lat()
    newLocation.longitude = place.geometry.location.lng()
    newLocation.geocodedAddress = place.formatted_address
    this.setState({
      contactData: { ...this.state.contactData, address: newLocation }
    })
    this.setState({
      autocomplete: { ...this.state.autocomplete, address: true }
    })
  }
  render () {
    const {
      theme,
      setContactAttempted,
      setContactBtn,
      t
    } = this.props
    const { contactData } = this.state

    return (
      <div className="flex offset-5 layout-row layout-wrap layout-align-start-start">
        <div className="flex-100 layout-row layout-wrap">
          <h3 className="flex-40">{t('user:addressDetails')}</h3>
          <div className="offset-5 flex-55">
            <GmapsWrapper
              theme={theme}
              component={PlaceSearch}
              inputStyles={{
                width: '96%',
                marginTop: '9px',
                background: 'white',
                boxShadow: 'unset'
              }}
              handlePlaceChange={e => this.handlePlaceChange(e)}
              hideMap
            />
          </div>
        </div>
        <div className="flex-100 layout-row">
          <IconLable faClass="map-marker" theme={theme} />
          <div className="flex-95 layout-row layout-wrap">
            <div className={`${styles.grouped_inputs} flex-100 layout-row`}>
              <FormsyInput
                wrapperClassName={`${styles.wrapper_input} flex-75`}
                className={styles.input}
                type="text"
                value={contactData.address.street}
                name="address-street"
                placeholder={t('user:street')}
                submitAttempted={setContactAttempted}
                errorMessageStyles={{
                  fontSize: '12px',
                  bottom: '-19px'
                }}
                validations="minLength:2"
                validationErrors={{
                  isDefaultRequiredValue: t('errors:twoChars'),
                  minLength: t('errors:twoChars')
                }}
                required
              />
              <FormsyInput
                wrapperClassName={`${styles.wrapper_input} offset-5 flex-15`}
                className={styles.input}
                type="text"
                value={contactData.address.streetNumber}
                name="address-streetNumber"
                placeholder={t('user:number')}
                submitAttempted={setContactAttempted}
                errorMessageStyles={{
                  fontSize: '12px',
                  bottom: '-19px'
                }}
                validationErrors={{
                  isDefaultRequiredValue: t('errors:notBlank')
                }}
                required
              />
            </div>
            <div className={`${styles.grouped_inputs} flex-100 layout-row`}>
              <FormsyInput
                wrapperClassName={`${styles.wrapper_input} flex-95`}
                className={styles.input}
                type="text"
                value={contactData.address.city}
                name="address-city"
                placeholder={t('user:city')}
                submitAttempted={setContactAttempted}
                errorMessageStyles={{
                  fontSize: '12px',
                  bottom: '-19px'
                }}
                validations="minLength:2"
                validationErrors={{
                  isDefaultRequiredValue: t('errors:twoChars'),
                  minLength: t('errors:twoChars')
                }}
                required
              />
            </div>
            <div className={`${styles.grouped_inputs} flex-100 layout-row`}>
              <FormsyInput
                wrapperClassName={`${styles.wrapper_input} flex-40`}
                className={styles.input}
                type="text"
                value={contactData.address.zipCode}
                name="address-zipCode"
                placeholder={t('user:postalCode')}
                submitAttempted={setContactAttempted}
                errorMessageStyles={{
                  fontSize: '12px',
                  bottom: '-19px'
                }}
                validations="minLength:4"
                validationErrors={{
                  isDefaultRequiredValue: t('errors:fourChars'),
                  minLength: t('errors:fourChars')
                }}
                required
              />
              <FormsyInput
                wrapperClassName={`${styles.wrapper_input} offset-5 flex-50`}
                className={styles.input}
                type="text"
                value={contactData.address.country}
                name="address-country"
                placeholder={t('user:country')}
                submitAttempted={setContactAttempted}
                errorMessageStyles={{
                  fontSize: '12px',
                  bottom: '-19px'
                }}
                validations="minLength:3"
                validationErrors={{
                  isDefaultRequiredValue: t('errors:threeChars'),
                  minLength: t('errors:threeChars')
                }}
                required
              />
            </div>
            <FormsyInput
              wrapperClassName="flex-100"
              type="hidden"
              value={contactData.address.geocodedAddress}
              name="address-geocodedAddress"
              placeholder=""
            />
            <div
              className={`${styles.grouped_inputs} flex-100 layout-row layout-align-start-start`}
              style={{ marginTop: '50px' }}
            >
              { setContactBtn }
            </div>
          </div>
        </div>
      </div>
    )
  }
}

AddressDetailsSection.propTypes = {
  theme: PropTypes.theme,
  t: PropTypes.func.isRequired,
  contactData: PropTypes.objectOf(PropTypes.any).isRequired,
  setContactAttempted: PropTypes.bool,
  setContactBtn: PropTypes.node
}

AddressDetailsSection.defaultProps = {
  theme: null,
  setContactAttempted: false,
  setContactBtn: null
}

export default withNamespaces(['errors', 'user'])(AddressDetailsSection)
