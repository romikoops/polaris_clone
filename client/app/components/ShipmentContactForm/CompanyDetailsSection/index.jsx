import React from 'react'
import { translate } from 'react-i18next'
import PropTypes from '../../../prop-types'
import FormsyInput from '../../FormsyInput/FormsyInput'
import styles from '../ShipmentContactForm.scss'
import IconLable from '../IconLable'

function CompanyDetailsSection ({
  theme, contactData, setContactAttempted, checkValid, t
}) {
  return (
    <div className="flex-100 layout-row layout-wrap">
      <h3>{t('user:companyDetails')}</h3>
      <div className={`${styles.grouped_inputs} flex-100 layout-row layout-wrap`}>
        <div className="flex-95 layout-row">
          <IconLable faClass="building-o" theme={theme} />
          <FormsyInput
            wrapperClassName={`${styles.wrapper_input} flex`}
            className={styles.input}
            type="text"
            value={contactData.contact.companyName}
            name="companyName"
            placeholder={t('user:companyName')}
            submitAttempted={setContactAttempted}
            errorMessageStyles={{
              fontSize: '12px',
              bottom: '-19px'
            }}
            validations="minLength:2"
            validationErrors={{
              isDefaultRequiredValue: t('errors:twoChars'),
              minLength: t('errors:twoChars')
            }}
            required
          />
        </div>
      </div>
      <div className={`${styles.grouped_inputs} flex-100 layout-row layout-wrap`}>
        <div className="flex-95 layout-row">
          <IconLable faClass="user" theme={theme} />
          <div className="flex layout-row layout-wrap">
            <FormsyInput
              wrapperClassName={`${styles.wrapper_input} flex-100`}
              className={styles.input}
              type="text"
              value={contactData.contact.firstName}
              name="firstName"
              placeholder={t('user:firstName')}
              submitAttempted={setContactAttempted}
              errorMessageStyles={{
                fontSize: '12px',
                bottom: '-19px'
              }}
              validations="minLength:2"
              validationErrors={{
                isDefaultRequiredValue: t('errors:twoChars'),
                minLength: t('errors:twoChars')
              }}
              required
            />
            <FormsyInput
              wrapperClassName={`${styles.wrapper_input} flex-100`}
              className={styles.input}
              type="text"
              value={contactData.contact.lastName}
              name="lastName"
              placeholder={t('user:lastName')}
              submitAttempted={setContactAttempted}
              errorMessageStyles={{
                fontSize: '12px',
                bottom: '-19px'
              }}
              validations="minLength:2"
              validationErrors={{
                isDefaultRequiredValue: t('errors:twoChars'),
                minLength: t('errors:twoChars')
              }}
              required
            />
          </div>
        </div>
      </div>
      <div className={`${styles.grouped_inputs} flex-100 layout-row layout-wrap`}>
        <div className="flex-95 layout-row">
          <IconLable faClass="envelope" theme={theme} />
          <FormsyInput
            wrapperClassName={`${styles.wrapper_input} flex-95`}
            className={styles.input}
            type="text"
            value={contactData.contact.email}
            name="email"
            placeholder={t('user:email')}
            onBlur={() => checkValid('email')}
            submitAttempted={setContactAttempted}
            errorMessageStyles={{
              fontSize: '12px',
              bottom: '-19px'
            }}
            validations={{
              matchRegexp: /^([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})$/i
            }}
            validationErrors={{
              isDefaultRequiredValue: t('errors:notBlank'),
              matchRegexp: t('errors:invalidEmail')
            }}
            required
          />
        </div>
        <div className="flex-95 layout-row">
          <IconLable faClass="phone" theme={theme} />
          <FormsyInput
            wrapperClassName={`${styles.wrapper_input} flex-95`}
            className={styles.input}
            type="text"
            value={contactData.contact.phone}
            name="phone"
            placeholder={t('user:phone')}
            submitAttempted={setContactAttempted}
            errorMessageStyles={{
              fontSize: '12px',
              bottom: '-19px'
            }}
            validations="minLength:4"
            validationErrors={{
              isDefaultRequiredValue: t('errors:fourChars'),
              minLength: 'Minimum 4 characters'
            }}
            required
          />
        </div>
      </div>
    </div>
  )
}

CompanyDetailsSection.propTypes = {
  theme: PropTypes.theme,
  t: PropTypes.func.isRequired,
  contactData: PropTypes.objectOf(PropTypes.any).isRequired,
  setContactAttempted: PropTypes.bool,
  checkValid: PropTypes.func.isRequired
}

CompanyDetailsSection.defaultProps = {
  theme: null,
  setContactAttempted: false
}

export default translate(['errors', 'user'])(CompanyDetailsSection)
