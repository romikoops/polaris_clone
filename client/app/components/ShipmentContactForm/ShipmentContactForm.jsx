import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Formsy from 'formsy-react'
import styles from './ShipmentContactForm.scss'
import { RoundButton } from '../RoundButton/RoundButton'
import { nameToDisplay } from '../../helpers'
import AddressDetailsSection from './AddressDetailsSection'
import CompanyDetailsSection from './CompanyDetailsSection'

export class ShipmentContactForm extends Component {
  static mapInputs (inputs) {
    const location = {}
    const contact = {}
    Object.keys(inputs).forEach((k) => {
      if (k.split('-')[0] === 'location') {
        location[k.split('-')[1]] = inputs[k]
      } else {
        contact[k] = inputs[k]
      }
    })
    return { location, contact }
  }
  constructor (props) {
    super(props)
    this.state = {
      contactData: { contact: {}, location: {} },
      setContactAttempted: false
    }
    this.handleFormChange = this.handleFormChange.bind(this)
    this.handleInvalidSubmit = this.handleInvalidSubmit.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleFormChange (event) {
    this.props.handleChange(event)
  }
  handlePlaceChange (place) {
    const newLocation = {
      streetNumber: '',
      street: '',
      zipCode: '',
      city: '',
      country: ''
    }
    place.address_components.forEach((ac) => {
      if (ac.types.includes('street_number')) {
        newLocation.streetNumber = ac.long_name
      }

      if (ac.types.includes('route') || ac.types.includes('premise')) {
        newLocation.street = ac.long_name
      }

      if (ac.types.includes('administrative_area_level_1') || ac.types.includes('locality')) {
        newLocation.city = ac.long_name
      }

      if (ac.types.includes('postal_code')) {
        newLocation.zipCode = ac.long_name
      }

      if (ac.types.includes('country')) {
        newLocation.country = ac.long_name
      }
    })
    newLocation.latitude = place.geometry.location.lat()
    newLocation.longitude = place.geometry.location.lng()
    newLocation.geocodedAddress = place.formatted_address
    this.setState({
      contactData: { ...this.state.contactData, location: newLocation }
    })
    this.setState({
      autocomplete: { ...this.state.autocomplete, location: true }
    })
  }
  handleSubmit (contactData) {
    this.props.setContact(contactData)
    this.contactForm.reset()
    this.setState({ setContactAttempted: false })
  }
  handleInvalidSubmit () {
    this.setState({ setContactAttempted: true })
  }

  render () {
    const { theme, contactType } = this.props
    const { contactData } = this.state

    const setContactBtn = (
      <RoundButton
        text={
          `${contactType === 'notifyee' ? 'Add' : 'Set'} ` +
          `${nameToDisplay(contactType)}`
        }
        theme={theme}
        size="small"
        active
      />
    )

    const addressDetailsSection = (
      <AddressDetailsSection
        theme={theme}
        contactData={contactData}
        handlePlaceChange={place => this.handlePlaceChange(place)}
        setContactAttempted={this.state.setContactAttempted}
        setContactBtn={setContactBtn}
      />
    )

    const setContactBtnWrapper = (
      <div
        className="flex-100 layout-row layout-align-center-center"
      >
        { setContactBtn }
      </div>
    )
    return (
      <div className={
        `${styles.wrapper_form} flex-100 ` +
        'layout-row layout-wrap layout-align-center-start'
      }
      >
        <Formsy
          className="flex-100 layout-row layout-wrap layout-align-start-start"
          name="form"
          onValidSubmit={this.handleSubmit}
          onInvalidSubmit={this.handleInvalidSubmit}
          mapping={ShipmentContactForm.mapInputs}
          ref={(c) => { this.contactForm = c }}
          style={{ padding: '8px 20px' }}
        >
          <div className={`flex${contactType === 'notifyee' ? '-100' : ''} layout-row`}>
            <CompanyDetailsSection
              theme={theme}
              contactData={contactData}
              setContactAttempted={this.state.setContactAttempted}
            />
          </div>

          { contactType === 'notifyee' ? setContactBtnWrapper : addressDetailsSection }
        </Formsy>
      </div>
    )
  }
}

ShipmentContactForm.propTypes = {
  theme: PropTypes.theme,
  setContact: PropTypes.func,
  handleChange: PropTypes.func,
  contactType: PropTypes.string
}

ShipmentContactForm.defaultProps = {
  theme: null,
  handleChange: null,
  setContact: null,
  contactType: ''
}

export default ShipmentContactForm
