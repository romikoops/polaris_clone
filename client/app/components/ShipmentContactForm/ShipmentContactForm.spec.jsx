import * as React from 'react'
import { shallow } from 'enzyme'
import { theme, identity, location } from '../../mocks'

jest.mock('formsy-react', () => {
  // eslint-disable-next-line
  const Formsy = ({ children }) => <div>{children}</div>

  return {
    default: Formsy
  }
})
jest.mock('../../hocs/GmapsWrapper', () => {
  // eslint-disable-next-line
  const GmapsWrapper = ({ children }) => <div>{children}</div>

  return {
    default: GmapsWrapper
  }
})
jest.mock('../Maps/PlaceSearch', () => {
  // eslint-disable-next-line
  const PlaceSearch = ({ children }) => <div>{children}</div>

  return {
    default: PlaceSearch
  }
})
jest.mock('../FormsyInput/FormsyInput', () => {
  // eslint-disable-next-line
  const FormsyInput = props => <input {...props} />

  return {
    default: FormsyInput
  }
})

// eslint-disable-next-line
import { ShipmentContactForm } from './ShipmentContactForm'

const propsBase = {
  theme,
  close: identity,
  setContact: identity,
  handleChange: identity,
  contactData: {
    location,
    contact: { companyName: 'FOO_COMPANY' },
    type: 'FOO_TYPE'
  }
}

const createShallow = propsInput => shallow(<ShipmentContactForm {...propsInput} />)

test('shallow rendering', () => {
  expect(createShallow(propsBase)).toMatchSnapshot()
})
