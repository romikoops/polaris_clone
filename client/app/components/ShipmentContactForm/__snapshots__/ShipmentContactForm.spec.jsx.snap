// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contactType !== notifyee 1`] = `
<div
  className="wrapper_form flex-100 layout-row layout-wrap layout-align-center-start"
>
  <Formsy
    className="flex-100 layout-row layout-wrap layout-align-start-start"
    disabled={false}
    getErrorMessage={[Function]}
    getErrorMessages={[Function]}
    getValue={[Function]}
    hasValue={[Function]}
    isFormDisabled={[Function]}
    isFormSubmitted={[Function]}
    isPristine={[Function]}
    isRequired={[Function]}
    isValid={[Function]}
    isValidValue={[Function]}
    mapping={[Function]}
    name="form"
    onChange={[Function]}
    onError={[Function]}
    onInvalid={[Function]}
    onInvalidSubmit={[Function]}
    onReset={[Function]}
    onSubmit={[Function]}
    onValid={[Function]}
    onValidSubmit={[Function]}
    preventExternalInvalidation={false}
    resetValue={[Function]}
    setValidations={[Function]}
    setValue={[Function]}
    showError={[Function]}
    showRequired={[Function]}
    style={
      Object {
        "padding": "8px 20px",
      }
    }
    validationErrors={null}
  >
    <div
      className="flex layout-row"
    >
      <CompanyDetailsSection
        checkValid={[Function]}
        contactData={Object {}}
        handleInputChange={[Function]}
        setContactAttempted={false}
        showEdit={false}
        t={[Function]}
        theme={
          Object {
            "colors": Object {
              "primary": "#333",
              "secondary": "#fafafa",
            },
          }
        }
      />
    </div>
    <AddressDetailsSection
      contactData={Object {}}
      handlePlaceChange={[Function]}
      newLocation={Object {}}
      setContactAttempted={false}
      setContactBtn={
        <RoundButton
          active={true}
          back={false}
          classNames=""
          disabled={false}
          flexContainer="100"
          handleDisabled={null}
          handleNext={null}
          icon=""
          iconClass=""
          inverse={false}
          size="small"
          text="Set foo"
          theme={
            Object {
              "colors": Object {
                "primary": "#333",
                "secondary": "#fafafa",
              },
            }
          }
        />
      }
      t={[Function]}
      theme={
        Object {
          "colors": Object {
            "primary": "#333",
            "secondary": "#fafafa",
          },
        }
      }
    />
  </Formsy>
</div>
`;

exports[`shallow rendering 1`] = `
<div
  className="wrapper_form flex-100 layout-row layout-wrap layout-align-center-start"
>
  <Formsy
    className="flex-100 layout-row layout-wrap layout-align-start-start"
    disabled={false}
    getErrorMessage={[Function]}
    getErrorMessages={[Function]}
    getValue={[Function]}
    hasValue={[Function]}
    isFormDisabled={[Function]}
    isFormSubmitted={[Function]}
    isPristine={[Function]}
    isRequired={[Function]}
    isValid={[Function]}
    isValidValue={[Function]}
    mapping={[Function]}
    name="form"
    onChange={[Function]}
    onError={[Function]}
    onInvalid={[Function]}
    onInvalidSubmit={[Function]}
    onReset={[Function]}
    onSubmit={[Function]}
    onValid={[Function]}
    onValidSubmit={[Function]}
    preventExternalInvalidation={false}
    resetValue={[Function]}
    setValidations={[Function]}
    setValue={[Function]}
    showError={[Function]}
    showRequired={[Function]}
    style={
      Object {
        "padding": "8px 20px",
      }
    }
    validationErrors={null}
  >
    <div
      className="flex-100 layout-row"
    >
      <CompanyDetailsSection
        checkValid={[Function]}
        contactData={Object {}}
        handleInputChange={[Function]}
        setContactAttempted={false}
        showEdit={false}
        t={[Function]}
        theme={
          Object {
            "colors": Object {
              "primary": "#333",
              "secondary": "#fafafa",
            },
          }
        }
      />
    </div>
    <div
      className="flex-100 layout-row layout-align-center-center"
    >
      <RoundButton
        active={true}
        back={false}
        classNames=""
        disabled={false}
        flexContainer="100"
        handleDisabled={null}
        handleNext={null}
        icon=""
        iconClass=""
        inverse={false}
        size="small"
        text="Add notifyee"
        theme={
          Object {
            "colors": Object {
              "primary": "#333",
              "secondary": "#fafafa",
            },
          }
        }
      />
    </div>
  </Formsy>
</div>
`;

exports[`showEdit is true 1`] = `
<div
  className="wrapper_form flex-100 layout-row layout-wrap layout-align-center-start"
>
  <Formsy
    className="flex-100 layout-row layout-wrap layout-align-start-start"
    disabled={false}
    getErrorMessage={[Function]}
    getErrorMessages={[Function]}
    getValue={[Function]}
    hasValue={[Function]}
    isFormDisabled={[Function]}
    isFormSubmitted={[Function]}
    isPristine={[Function]}
    isRequired={[Function]}
    isValid={[Function]}
    isValidValue={[Function]}
    mapping={[Function]}
    name="form"
    onChange={[Function]}
    onError={[Function]}
    onInvalid={[Function]}
    onInvalidSubmit={[Function]}
    onReset={[Function]}
    onSubmit={[Function]}
    onValid={[Function]}
    onValidSubmit={[Function]}
    preventExternalInvalidation={false}
    resetValue={[Function]}
    setValidations={[Function]}
    setValue={[Function]}
    showError={[Function]}
    showRequired={[Function]}
    style={
      Object {
        "padding": "8px 20px",
      }
    }
    validationErrors={null}
  >
    <div
      className="flex-100 layout-row"
    >
      <CompanyDetailsSection
        checkValid={[Function]}
        contactData={Object {}}
        handleInputChange={[Function]}
        setContactAttempted={false}
        showEdit={true}
        t={[Function]}
        theme={
          Object {
            "colors": Object {
              "primary": "#333",
              "secondary": "#fafafa",
            },
          }
        }
      />
    </div>
    <div
      className="flex-100 layout-row layout-align-center-center"
    >
      <RoundButton
        active={true}
        back={false}
        classNames=""
        disabled={false}
        flexContainer="100"
        handleDisabled={null}
        handleNext={null}
        icon=""
        iconClass=""
        inverse={false}
        size="small"
        text="Set notifyee"
        theme={
          Object {
            "colors": Object {
              "primary": "#333",
              "secondary": "#fafafa",
            },
          }
        }
      />
    </div>
  </Formsy>
</div>
`;
